app/models/service_template_ansible_playbook.rb

```diff
@@ -52,6 +52,15 @@ def self.create_catalog_item(options, auth_user)
     end
   end
 
+  def self.manager_valid?(manager)
```

Comment:
@bzwei - I went endwise around associations like 
`manager.configuration_script_sources.present? &&
manager.configuration_script_payloads.present? &&
manager.credentials.where(:type => ManageIQ::Providers::AnsibleTower::AutomationManager::MachineCredential).present?`

To mitigate queries calling `tablename.*`. I was worried about performance on this validation if there were large amounts of `playbooks`, `credentials`, `projects` on the Tower.

That being said - I didn't realize `.count` would rewrite the query and not do a ruby based count of the returned objects.  So - for performance I can change to your first suggestion.